lootbox/api.py

```diff
@@ -111,177 +135,314 @@ async def get_drop_handler(
-#         )
-
-#     return data.ClaimantsResponse(claimants=results)
-
-
-# @app.delete("/drops/claimants", response_model=data.RemoveClaimantsResponse)
-# async def create_claimants(
-#     request: Request,
-#     remove_claimants_request: data.DropRemoveClaimantsRequest = Body(...),
-#     db_session: Session = Depends(db.yield_db_session),
-# ) -> data.RemoveClaimantsResponse:
-
-#     """
-#     Remove addresses to particular claim
-#     curl --location --request DELETE 'localhost:7191/drops/claimants' \
-#     --header 'Content-Type: application/json' \
-#     --data-raw '{"dropper_claim_id": UUID, "claimants":[address]}'
-#     """
-
-#     try:
-#         results = actions.delete_claimants(
-#             db_session=db_session,
-#             dropper_claim_id=remove_claimants_request.dropper_claim_id,
-#             addresses=remove_claimants_request.addresses,
-#         )
-#     except Exception as e:
-#         raise DropperHTTPException(
-#             status_code=500, detail=f"Error removing claimants {e}"
-#         )
-
-#     return data.RemoveClaimantsResponse(addresses=results)
+        logger.error(
+            f"Can't get claims for user {claimant_address} end with error: {e}"
+        )
+        raise DropperHTTPException(status_code=500, detail="Can't get claims")
+
+    return data.DropListResponse(drops=[result for result in results])
+
+
+@app.get("/drops/terminus/claims", response_model=data.DropListResponse)
+async def get_drop_list_handler(
+    blockchain: str,
+    terminus_address: str,
+    terminus_pool_id: int,
+    dropper_contract_address: Optional[str] = Query(None),
+    active: Optional[bool] = Query(None),
+    limit: int = 10,
+    offset: int = 0,
+    db_session: Session = Depends(db.yield_db_session),
+) -> data.DropListResponse:
+    """
+    Get list of drops for a given dropper contract and claimant address.
+    """
+
+    if dropper_contract_address:
+        dropper_contract_address = Web3.toChecksumAddress(dropper_contract_address)
+
+    terminus_address = Web3.toChecksumAddress(terminus_address)
+
+    try:
+        results = actions.get_terminus_claims(
+            db_session=db_session,
+            dropper_contract_address=dropper_contract_address,
+            blockchain=blockchain,
+            terminus_address=terminus_address,
+            terminus_pool_id=terminus_pool_id,
+            active=active,
+            limit=limit,
+            offset=offset,
+        )
+    except NoResultFound:
+        raise DropperHTTPException(status_code=404, detail="No drops found.")
+    except Exception as e:
+        logger.error(
+            f"Can't get Terminus claims (blockchain={blockchain}, address={terminus_address}, pool_id={terminus_pool_id}): {e}"
+        )
+        raise DropperHTTPException(status_code=500, detail="Can't get claims")
+
+    return data.DropListResponse(drops=[result for result in results])
+
+
+@app.post("/drops/claims", response_model=data.DropCreatedResponse)
```

Comment:
Add authentication here - user address must match address of dropper contract owner.